app.js

```diff
@@ -1,22 +1,35 @@
 const express = require("express")
 const  { 
     getApi,
-    getTopics
+    getTopics,
+    getArticle
 } = require('./controllers/api.controller')
 
 const app = express()
-app.use(express.json())
 
 app.get('/api', getApi)
 app.get('/api/topics', getTopics)
+app.get('/api/articles/:article_id', getArticle)
+
 
 app.use((err, req, res, next) => {
- if (err.status && err.msg) {
-    res.status(err.status).send({ msg: err.msg })
- } else {
-    next(err)
- }
+    console.log(err, "<---error in 1st app.use")
+    if (err.status && err.msg) {
+        res.status(err.status).send({ msg: err.msg })
+    } else {
+        next(err)
+    }
 })
+
+app.use((err, req, res, next) => {
+    console.log(err.code, "err.code in 2nd app.use")
+    if (err.code === "22P02") {
+        res.status(400).send({ msg: "Bad request" })
+    } else {
+        next(err)
+    }
+})
+
```

Comment:
As you've split these up, it would make sense to turn them into specific error handling functions to better follow your own code.